diff --git a/addon/instance-initializers/clear-double-boot.js b/addon/instance-initializers/clear-double-boot.js
index fd082ea76f899155c94b21ad299cce80c248c444..2627eb20726d6700123c5f1207f266a43d9b552e 100644
--- a/addon/instance-initializers/clear-double-boot.js
+++ b/addon/instance-initializers/clear-double-boot.js
@@ -35,11 +35,11 @@ export default {
 
   initialize(instance) {
     if (typeof FastBoot === 'undefined') {
-      var originalDidCreateRootView = instance.didCreateRootView;
+      var originalRenderRootComponent = instance.renderRootComponent;
 
-      instance.didCreateRootView = function () {
+      instance.renderRootComponent = function () {
         clearHtml();
-        originalDidCreateRootView.apply(instance, arguments);
+        return originalRenderRootComponent.apply(instance, arguments);
       };
     }
   },
diff --git a/fastboot/initializers/ajax.js b/fastboot/initializers/ajax.js
index bd19a93992611f96931439129a37f02b08941678..9b817f41358602726bc486841e306390731012e4 100644
--- a/fastboot/initializers/ajax.js
+++ b/fastboot/initializers/ajax.js
@@ -1,8 +1,6 @@
 /* eslint-disable ember/new-module-imports */
 /* globals najax */
-import Ember from 'ember';
-
-const { get } = Ember;
+import { get } from '@ember/object';
 
 var nodeAjax = function (options) {
   let httpRegex = /^https?:\/\//;
diff --git a/fastboot/initializers/error-handler.js b/fastboot/initializers/error-handler.js
index 96177b2db336a5142588e10fddaa333fc7691883..d08265cdc9ed660009adcd72a20d24937ca58def 100644
--- a/fastboot/initializers/error-handler.js
+++ b/fastboot/initializers/error-handler.js
@@ -1,5 +1,3 @@
-import Ember from 'ember';
-
 /**
  * Initializer to attach an `onError` hook to your app running in fastboot. It catches any run loop
  * exceptions and other errors and prevents the node process from crashing.
@@ -8,13 +6,5 @@ import Ember from 'ember';
 export default {
   name: 'error-handler',
 
-  initialize: function () {
-    if (!Ember.onerror) {
-      // if no onerror handler is defined, define one for fastboot environments
-      Ember.onerror = function (err) {
-        const errorMessage = `There was an error running your app in fastboot. More info about the error: \n ${err.stack || err}`;
-        console.error(errorMessage);
-      };
-    }
-  },
+  initialize: function () {},
 };
